### PR TITLE
Extract background mode handling outside of the pod

### PR DESCRIPTION
**Pull request by samuel-gallet**
Otherwise, the pod does not build for app extension

### CODE REVIEW

File: MatrixSDK/Data/Store/MXFileStore/MXFileStore.m [hunk 1]
```diff
@@ -231,11 +230,11 @@ - (void)openWithCredentials:(MXCredentials*)someCredentials onComplete:(void (^)
         }
         
         dispatch_async(dispatch_get_main_queue(), ^{
-#if TARGET_OS_IPHONE
-            [[UIApplication sharedApplication] endBackgroundTask:backgroundTaskIdentifier];
-            backgroundTaskIdentifier = UIBackgroundTaskInvalid;
-#endif
-
+            id<MXBackgroundModeHandler> handler = [MXSDKOptions sharedInstance].backgroungModeHandler;
+            if (handler && backgroundTaskIdentifier == [handler invalidIdentifier]) {
```
Comment (manuroe) [inline MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: Could you please follow the coding rules of the rest of the project and put a EOL between ```if ()``` and ```{...```.

**Comment by samuel-gallet** [comment]
I have edited the commit.
I could not succeed to provide default implementation. If you do, you have to include
[UIApplication sharedApplication] into the SDK.

**Comment by samuel-gallet** [comment]
up

**Comment by manuroe** [comment]
Did you see my remark about backgroungModeHandler wrong typing?

Can you also add a default implementation?
For the moment, if it is better for you, exclude it from the swift pod like  in https://github.com/matrix-org/matrix-ios-sdk/pull/256/files#diff-1ed2b53be61c4d854c1ea132295c04acR25

My goal is to propose later this implementation as a subspec as we talked at https://github.com/matrix-org/matrix-ios-sdk/pull/298#issuecomment-306407010.

**Comment by samuel-gallet** [comment]
I'm sorry but I can't see your comment about backgroundModelHandler wrong typing

**Comment by manuroe** [comment]
Sorry. This comment was pending in a new review.
I do not know why GH did not publish it when I added it in my first review. Anyway, that should be ok now.

File: MatrixSDK/MXDefaultBackgroundModeHandler.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
+//
```
Comment (manuroe) [inline MatrixSDK/MXDefaultBackgroundModeHandler.h]: Could you please use the Apache header?
You can use your name in like https://github.com/matrix-org/matrix-ios-sdk/blob/master/MatrixSDK/SwiftMatrixSDK.h

File: MatrixSDK/MXDefaultBackgroundModeHandler.m [hunk 3]
```diff
@@ -0,0 +1,38 @@
+//
+//  MXDefaultBackgroundModeHandler.m
```
Comment (manuroe) [inline MatrixSDK/MXDefaultBackgroundModeHandler.m]: Apache header

File: MatrixSDK/MXDefaultBackgroundModeHandler.h [hunk 4]
```diff
@@ -0,0 +1,22 @@
+/*
+ Copyright 2017 Avery Pierce
```
Comment (manuroe) [inline MatrixSDK/MXDefaultBackgroundModeHandler.h]: Are you Avery?

**Pull request merged**